localization.go

```diff
@@ -0,0 +1,55 @@
+package sprig
+
+// Translator interface
+type Translator interface {
+	T(string) string
+}
+
+var locale = InitLocale("EN")
+
+// Localization type for serving the
+// localization functions,
+// implements Translator
+type Localization struct {
+	locale       string
+	dictionaries map[string]map[string]string
+}
+
+// Localization initialization
+func InitLocale(lang string) Localization {
+	var localization = Localization{}
+	localization.SetLocale(lang)
+
+	return localization
+}
+
+// Sets default locale
+func (l *Localization) SetLocale(lang string) {
+	l.locale = lang
+}
+
+// Returns default locale
+func (l *Localization) GetLocale() string {
+	return l.locale
+}
+
+// Sets localization dictionary
+func (l *Localization) SetTranslation(locale string, dictionary map[string]string) Localization {
+	if l.dictionaries == nil {
+		l.dictionaries = make(map[string]map[string]string)
+	}
+	l.dictionaries[locale] = make(map[string]string)
+	l.dictionaries[locale] = dictionary
+	return *l
+}
+
+// Translation function
+func (l *Localization) T(word string) string {
```

Comment:
T is great for dealing with strings. Some things have to do with numbers and plurals of words. For example it could be "0 something" vs "5 somethings". In addition to basic string translations what about plural handling as well?

Reply:
yes, the next step as i planned, if you accept my help, i add more functions such as handle the plural in different graps, something like this

`'apples' => '{0} There are none|[1,19] There are some|[20,*] There are many'`

in template would be 

`{{ t_choice 'apples' 10 }}`

Reply:
Btw, thank you for review, i will take care for it.